CenterForOpenScience/ember-osf-web
# [ENG-437] View only links

### NOTE: Not testable without BE changes (but could be safely merged/released without them) https://github.com/CenterForOpenScience/osf.io/pull/9020

## Purpose
Support view-only links.


## Summary of Changes
- Add `view_only` query param to application route
- Store VOL info on the `current-user` service
- Include `view_only` in all API requests, all links within OSF
- Display a banner with a button to stop using a VOL
- If the VOL is anonymized, display "Anonymous contributors" in place of any contributor list


## Side Effects
Added some VOL logic to the `@checkAuth` decorator/mixin because the VOL token should be set *before* the mixed-in `beforeModel` to correctly handle all auth flows. This violates `@checkAuth`'s encapsulation, but the need for this seems to indicate it shouldn't be a decorator/mixin at all. In the future, we should refactor it to a service method called in the application route's `beforeModel`


## Feature Flags



## QA Notes
Backend branch required for testing VOLs on the new registries overview page: https://github.com/CenterForOpenScience/osf.io/pull/9020

Registries scenarios:
- Non-anonymized VOL: anyone can view (even embargoed reg, even logged out)
- Anonymized VOL: anyone can view (even embargoed reg, even logged out), but no contributor info available -- contributor lists say "Anonymous contributors"

Other things:
- Because VOLs in the API are more comprehensive than in legacy, we always display a VOL banner. Clicking "Leave this view" in the banner should take you to the landing page (dashboard, if logged in) without `view_only` in the URL.
- When you start with a VOL and browse around OSF, `view_only=blahblahblah` should stay in the URL
- Anonymized VOLs hide *all* contributor info. If you start with an anonymized VOL and navigate e.g. to the dashboard, all projects in your dashboard list will be anonymized too.




## Ticket


https://openscience.atlassian.net/browse/ENG-437

# Reviewer Checklist
